Add support for May-June 2021 schedule

psvenk opened the pull request:
Fixes #288. This pull request adds support for the anticipated schedule for May and June 2021. Please note that the current support is preliminary and needs changes in the code.

psvenk committed:
schedule.toml: add Vim folding markers

Makes navigation easier in Vim (and possibly other editors)

psvenk committed:
schedule.toml: add May/June 2021 schedule

psvenk committed:
Use new schedule + improve frontend support

Re-introduce the lunch slider and make the update_lunch function work
properly

psvenk committed:
Remove lunch C

Reflect the fact that the new schedule has only two lunches in the lunch
slider.

psvenk committed:
Fix up lunch guessing temporarily

Instead of initially loading a lunch-agnostic schedule but showing lunch
A as selected in the slider, choose lunch A by default. This should be
changed to code that guesses the student's lunch.

psvenk commented:
I think this should be ready for review (and hopefully merge). As far as I remember, the only omission here is that lunch guessing has not been incorporated yet. Because we do not yet have information about how lunches will be assigned, we can address that in a separate pull request (to close #95) and merge this for now (which defaults to lunch A).

I'm not sure if this uses `localStorage` to store the user's lunch; someone (possibly me) should test that.

jadebuckwalter commented:
@psvenk We could temporarily create a lunch slider where people can choose between lunch A and lunch B (like in the past), but that might not be necessary if we implement lunch guessing as soon as we know how lunches are being assigned.

Also, since Wednesdays alternate between black and silver, how should we take that into account? Maybe instead of determining whether it's a black or silver week, we just display both classes (if there are different classes for black and silver days) in Wednesday's schedule and clock. We could also hard-code for the remainder of the year, but that's not the best solution.

psvenk committed:
Update May-June schedule timings

In accordance with recent changes from the school to make sure that all
blocks are 50 minutes long on Wednesdays (block 1 was 45 minutes long).

jadebuckwalter committed:
Add lunch slider

Move the lunch slider from the clock to the schedule tab and align it
with the day of the week dropdown. I also made it a bit smaller.

psvenk committed:
Properly default to lunch A

Prevent a race condition that causes the potentially confusing
lunch-agnostic schedule to display instead of lunch A by default.

psvenk committed:
Implement lunch guessing

In accordance with the new lunch A and lunch B assignments for the rest
of the year. Also make sure that the slider is updated accordingly when
a lunch is determined based on the period 3 class.

psvenk committed:
Add link to document detailing schedule irregularities

We cannot possibly handle all of the irregular schedules over the next
several weeks; luckily, Ms. Yu has come to the rescue and made this
helpful document.

jadebuckwalter committed:
Add "alternating silver/black" for Wednesdays

Hopefully this will clear up some confusion even if the schedule doesn't
adapt accordingly.

psvenk committed:
Fix passing times in schedule

psvenk committed:
Update lunch slider colors

Make the colors more adaptive for dark mode.

jadebuckwalter reviewed (approved):
Looks good to me!
